SwiftCompilerSources/Sources/Optimizer/Utilities/EscapeInfoVisitor.swift

```diff
@@ -0,0 +1,633 @@
+import SIL
+
+struct EscapeInfoState {
+  let followStores: Bool
+  let knownType: Type?
+  let walkDown: Bool
```

Comment:
`walkDown` shouldn't be a member of the state. It's only used as a return value of a visit function, but not as input when passing the state _to_ a visit function.

Instead, you can define a separate walk-result enum which has a `continueWalkingDown` case (as it is in the current EscapeInfo implementation).
This would also be nicer, because `abortWalk` should really be called `markEscaping` in the context of EscapeInfo